File: 2. Add Two Numbers/main.ts
function twoSum(nums: number[], target: number): number[] {
    const complementMap = new Map<number, number>();

    for (let i = 0; i < nums.length; i++) {
        const complement = target - nums[i];
        if (complementMap.has(complement)) {
            return [complementMap.get(complement)!, i];
        }
        complementMap.set(nums[i], i);
    }

    return [];
}

File: 2. Add Two Numbers/main.dart
class Solution {
  List<int> twoSum(List<int> nums, int target) {
    Map<int, int> complementMap = {};

    for (var i = 0; i < nums.length; i++) {
      final complement = target - nums[i];
      final int? index = complementMap[complement];
      if (index != null) {
        return [index, i];
      }
      complementMap[nums[i]] = i;
    }

    return new List.empty();
  }
}
